x-pack/plugins/task_manager/server/task_pool.ts

```diff
@@ -125,7 +126,17 @@ export class TaskPool {
     taskRunner
       .run()
       .catch(err => {
-        this.logger.warn(`Task ${taskRunner.toString()} failed in attempt to run: ${err.message}`);
+        // If a task Saved Object can't be found by an in flight task runner
+        // we asssume the underlying task has been deleted while it was running
+        // so we will log this as a debug, rather than a warn
+        const errorLogLine = `Task ${taskRunner.toString()} failed in attempt to run: ${
+          err.message
+        }`;
+        if (isTaskSavedObjectNotFoundError(err, taskRunner.id)) {
```

Comment:
I like these "warn or debug" flows, especially since it feels like we don't have enough debug logging to begin with :-)